streams/src/main/java/org/apache/kafka/streams/state/internals/CachingWindowStore.java

```diff
@@ -270,4 +285,136 @@ public void close() {
         cache.close(name);
         wrapped().close();
     }
+
+    private class CacheIteratorWrapper implements PeekingKeyValueIterator<Bytes, LRUCacheEntry> {
```

Comment:
Are these two iterator wrapper's sharing the same logic?

Reply:
I believe they share enough logic that they *could* be consolidated, however I decided to leave them separate for now for a few reasons:
1) Cleaner as inner classes since they need access to a number of the outer class's members
2) Easier to reason about the logic for each in the context of the specific store, with variable names that reflect the appropriate situation
3) There may be additional optimizations we want to consider, which may have differing logic

None of these are necessarily reasons to *never* consolidate them, just to defer that for now

Reply:
Thanks, makes sense to me.